Assets/Scripts/HandController.cs

```diff
@@ -4,36 +4,53 @@
 
 public class HandController : MonoBehaviour
 {
+    public List<GameObject> cards;
+    public List<GameObject> selectedCards;
+    public int maxCards;
     public GameObject cardPrefab;
-    public GameObject[] cards;
-    public int numCards;
 
-    // Start is called before the first frame update
+    public CastButtonController castButton;
+
     void Start()
     {
         GenerateCards();
     }
 
+    public void SelectCard(GameObject card) {
+        if (!selectedCards.Contains(card)) {
```

Comment:
Actually I would use a hashset if you are going to have a second data structure to keep track of cards that have been selected for O(1) searching, otherwise theres no diff than just iterating through the cards you have and checking to see if it exists.

Reply:
^^ As above. If we end up moving the cards elsewhere, we would probably want to use List to preserve the order. For the code right now, though, HashSet is definitely better.

I'm going to leave this as List for now.